ui-core/src/components/inputs/datePicker/useDatePicker.ts

```diff
@@ -0,0 +1,78 @@
+import { useState, useRef, useEffect } from 'react';
+import useClickOutside from '../../hooks/useOutsideClick';
+import { useModalPosition } from '../../hooks/useModalPosition';
+import { CalendarSlot } from './type';
+import { formatDateString } from './utils';
+import { DatePickerProps } from './DatePicker';
+
+const MODAL_HORIZONTAL_OFFSET = -24;
+const MODAL_VERTICAL_OFFSET = 3;
+
+export default function useDatePicker({ calendarPickerProps, inputProps }: DatePickerProps) {
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const calendarPickerRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  useClickOutside(calendarPickerRef, (e) => {
+    // Do not close the picker if any children in input wrapper is clicked.
+    // This wrapper include, the input itself, the trailing content (which contains the calendar icon) and the leading content
+    if (inputRef.current && inputRef.current.parentElement?.contains(e.target as Node)) return;
+    setShowPicker(false);
+  });
+  const { calculatePosition, modalPosition } = useModalPosition(
+    inputRef,
+    calendarPickerRef,
+    MODAL_VERTICAL_OFFSET,
+    MODAL_HORIZONTAL_OFFSET
+  );
+
+  const formatSlotToInputValue = (newSelectedSlot: CalendarSlot | undefined) => {
+    let formattedDate = '';
+
+    if (newSelectedSlot) {
+      const { start, end } = newSelectedSlot;
+      if (calendarPickerProps.isRangeMode) {
+        if (start && end) {
+          formattedDate =
+            start.getFullYear() === end.getFullYear()
+              ? `${formatDateString(start).slice(0, 5)} - ${formatDateString(end)}`
+              : `${formatDateString(start)} - ${formatDateString(end)}`;
+        } else if (start) {
+          formattedDate = `${formatDateString(start)} -`;
+        }
```

Comment:
It's only a suggestion, as you prefer
```suggestion
      if (start) {
        let formattedStart = formatDateString(start);
        if (end && start.getFullYear() === end.getFullYear()) {
          formattedStart = formattedStart.slice(0, 5);
        }
        formattedDate = `${formattedStart} - ${formatDateString(end)}`;
      }
```